feat: unify events (#255)
emdgroup-liquid/liquid: src/docs/pages/form-validation.md

---
eleventyNavigation:
  key: Form validation
  parent: Introduction
  order: 8
layout: layout.njk
title: Form validation
permalink: introduction/form-validation/
---

# Form validation

> Liquid Oxygen wants to make integrating form components with forms and form validation libraries as easy as possible. There are mainly three possible ways of accessing the values of form-related components:

Reviewer: ```suggestion
Liquid Oxygen aims to make integrating form components with forms and form validation libraries as easy as possible. There are mainly three possible ways of accessing the values of form-related components:
```

- Integrating the Web Components with an associated `<form>` element and accessing their values using the `FormData` of the form
- Directly accessing the props of a Web Component
- Using the `detail` attribute of a custom event dispatched by the Web Component

## Form association

> There are two ways of associating a form-related with an existing `<form>` element:

Reviewer: ```suggestion
There are two ways of associating a form-related Web Component with an existing `<form>` element:
```

  function validateInput1() {
    const formData = new FormData(form)

    if (!formData.get('firstName')) {
      inputField1.invalid = true
      inputMessage1.style.visibility = 'inherit'
      return false
    }

    inputField1.invalid = false
    inputMessage1.style.visibility = 'hidden'
    return true
  }

  inputField1.addEventListener('input', validateInput1)

  inputField1.addEventListener('blur', validateInput1)

  function getValues() {
    const formData = new FormData(form)
    let values = '';

    formData.forEach((value, key) => {
      values += `
${key}: ${value}`
    })

    return values
  }

  form.addEventListener('submit', event => {
    event.preventDefault()
    const isValid = validateInput1()
    const values = getValues()

    // setTimeout is used in order to let the style update before showing the alert.
    // You can probably dismiss the timeout, if you're using a UI framework like React or Vue.
    setTimeout(() => {
      if (isValid) {
        window.alert(`Form submitted. FormData:${values}`)
      } else {
        window.alert('Form is invalid.')
      }
    }, 100)
  })
</script>
{% endexample %}

## Accessing props

> Accessing props works just as it would with native input elements. The props that a Liquid Oxygen component holds its content in can be found on the component's documentation page.

Reviewer: ```suggestion
Accessing props works just as it would with native input elements. The props, which allow you to access component specific values, are documented in each component's documentation page.
```
